gst/nnstreamer/nnstreamer_watchdog.h

```diff
@@ -0,0 +1,52 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * NNStreamer watchdog header
+ * Copyright (C) 2024 Gichan Jang <gichan2.jang@samsung.com>
+ */
+/**
+ * @file	nnstreamer_watchdog.h
+ * @date	30 Oct 2024
+ * @brief	NNStreamer watchdog header to manage the schedule in the element.
+ * @see		https://github.com/nnstreamer/nnstreamer
+ * @author	Gichan Jang <gichan2.jang@samsung.com>
+ * @bug		No known bugs except for NYI items
+ */
+
+
+#ifndef __NNSTREAMER_WATCHDOG_H__
+#define __NNSTREAMER_WATCHDOG_H__
+
+#include <glib.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef void *nns_watchdog_h;
+
+/**
+ * @brief Create nnstreamer watchdog. Recommended using watchdog handle with proper lock (e.g., GST_OBJECT_LOCK())
+ */
+void nnstreamer_watchdog_create (nns_watchdog_h *watchdog_h);
```

Comment:
How about changing the return type to check if the API is successful?

Reply:
Changed to return result. Thanks.

Reply:
Thank you for a great PR :+1: